niworkflows/interfaces/images.py

```diff
@@ -392,11 +393,26 @@ def _run_interface(self, runtime):
         # Rows 3-4:
         # Note: if qform is not valid, matching_affines is False
         elif (valid_sform and sform_code > 0) and (not matching_affines or qform_code == 0):
-            img.set_qform(img.get_sform(), sform_code)
+            img.set_qform(sform, sform_code)
+            new_qform = img.get_qform()
+            if np.allclose(new_qform, qform) and qform_code > 0:
+                # False alarm
+                self._results['out_file'] = self.inputs.in_file
+                open(out_report, 'w').close()
+                self._results['out_report'] = out_report
+                return runtime
+            diff = np.linalg.inv(qform) @ new_qform
+            trans, rot, scale, shear = transforms3d.affines.decompose44(diff)
```

Comment:
(from codacy, unused variable `shear`):

```suggestion
            trans, rot, scale, _ = transforms3d.affines.decompose44(diff)
```